src/scripts/components/colour/Panels.js

```diff
@@ -67,14 +70,22 @@ export default class Panels extends Component {
           showAllApps   : settings.showAllApps,
           showWebStore  : settings.showWebStore,
           showShortcuts : settings.panelShortcuts,
-          showFavicons  : settings.showFavicons
+          showFavicons  : settings.showFavicons,
+          blacklist     : settings.blacklist
         });
-
         if (settings.panelVisited) {
           Chrome.getTopSites(settings.maxVisited)
             .then((items) => {
+              let filtered = [];
+
+              for(var i = 0; i < items.length; i++){
+                if(!settings.blacklist[items[i].url]){
+                  filtered.push(items[i])
+                }
+              }
+
               this.setState({
-                topSites: items
+                topSites: filtered
```

Comment:
Could probably do `items.filter((item) => !settings.blacklist[item.irl])` or something instead of the forloop above.